### ISSUE TITLE
TM 4.16 fails to load script (too much recursion)

### ISSUE THREAD

**Issue by McBen** [Tampermonkey/tampermonkey#1473]
## Actual Behavior
With the new v4.16.6160 TM is no longer able to load my script.
Browser console window lists these error messages:
```
Unchecked lastError value: Error: too much recursion [common.js:4](moz-extension://2649.../rea/common.js)
    sendMessage moz-extension://2649.../rea/common.js:4
```

After that the page becomes unusable slow and none of the script are loaded.

## Specifications
- Firefox: 100.0b2
- TM: 4.16.6160

## Script
it's a (bigger) obfuscated script:
[blabla.user.zip](https://github.com/Tampermonkey/tampermonkey/files/8429097/blabla.user.zip)

**Comment by JamYiz** [comment]
I also have this problem in Firefox, please fix it in time

**Comment by McBen** [comment]
on Chrome it increases page loading time to +2min

**Comment by McBen** [comment]
looks like related to #1478

**Comment by zerah54** [comment]
Don't know if it can help, but as I still have the issue (#1474), I tried to find some way to make my script works.
Finally, if I remove "use strict" and the line below, my script loads well. 

`// @require      http://html2canvas.hertzen.com/dist/html2canvas.min.js`

Each line removed alone didn't solve the problem, but remove both seems to work.

**Comment by derjanb** [comment]
Can you please check whether the issue is fixed at 4.17.6161 (i[xpi](https://firefox.tampermonkey.net/firefox-current-beta.xpi))?

Please install the [BETA version](https://www.tampermonkey.net/index.php?browser=firefox) or check for BETA version updates at `about:addons` 

To test the fix please export your settings and scripts as zip or (JSON) file at the "Utilities" tab and import it back at the fixed BETA version.

**Comment by McBen** [comment]
> Can you please check whether the issue is fixed at 4.17.6161 (i[xpi](https://firefox.tampermonkey.net/firefox-current-beta.xpi))?

looks okay for me
